source/iface/usm_basic_concept.rst

```diff
@@ -2,288 +2,317 @@
-                       const sycl::property_list &propList = {})
-   template <typename T>
-   T* aligned_alloc(size_t alignment,
-                    size_t count,
-                    const sycl::device& syclDevice,
-                    const sycl::context& syclContext,
-                    sycl::usm::alloc kind,
-                    const sycl::property_list &propList = {})
-   void *aligned_alloc(size_t alignment,
-                       size_t numBytes,
-                       const sycl::queue& syclQueue,
-                       sycl::usm::alloc kind,
-                       const sycl::property_list &propList = {})
-   template <typename T>
-   T* aligned_alloc(size_t alignment,
-                    size_t count,
-                    const sycl::queue& syclQueue,
-                    sycl::usm::alloc kind,
-                    const sycl::property_list &propList = {})
-
-.. rubric:: Parameters
-
-==================  ===
-``alignment``       alignment of allocated data
-``numBytes``        allocation size in bytes
-``count``           number of elements
-``syclDevice``      See :ref:`device`
-``syclQueue``       See :ref:`queue`
-``syclContext``     See :ref:`context`
-``kind``            See :ref:`usm-alloc`
-``propList``
-==================  ===
-
-.. seealso:: `SYCL Specification <https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#_parameterized_allocation_functions>`__
-
-
-
-.. _sycl-free:
-
-==============
-``sycl::free``
-==============
+.. _usm-types:
+
+============
+Kinds of USM
+============
+
+USM builds upon Unified Addressing to define a shared
+address space where pointer values in this space always
+refer to the same location in memory. USM defines three
+types of memory allocations: ``host``, ``device``, and ``shared``.
+
+The following ``enum`` is used to refer to the different
+types of allocations inside of a SYCL program:
 
 ::
 
-   void free(void* ptr, sycl::context& context);
-   void free(void* ptr, sycl::queue& q);
+  namespace sycl::usm {
+
+  enum class alloc : /* unspecified */ {
+    host,
+    device,
+    shared,
+    unknown
+  };
+
+  } // namespace sycl::usm
+
+.. list-table::
+  :header-rows: 1
+
+  * - USM allocation type
+    - Description
+  * - ``host``
+    - Allocations in host memory that are accessible by a device.
+  * - ``device``
+    - Allocations in device memory that are not accessible by the host.
+  * - ``shared``
+    - Allocations in shared memory that are accessible by both host and device.
+
+USM is an optional feature which may not be supported by all devices, and
+devices that support USM may not support all types of USM allocation.
+
+A SYCL application can use the ``sycl::device::has()``
+function to determine the level of USM support for a device
+(See :ref:`device-aspects`).
+
+.. seealso:: |SYCL_SPEC_USM_KINDS|
+
+Characteristics of the different kinds of USM allocation
+--------------------------------------------------------
+
+.. list-table::
```

Comment:
I'd recommend removing this table from the reference guide.

Reply:
Removed in the latest commit. 

I also renamed the section from "Characteristics of the different kinds of USM allocation" to  "Rules of USM allocations". Can you suggest maybe another title for it?

Reply:
I suggest naming the section "USM accesses must be in the correct context"

Reply:
Ok, for now we'll name it this way, if everyone's fine with it